### ISSUE TITLE
Cannot use dump-to-h2 with MB_DB_CONNECTION_URI env var

### ISSUE THREAD

**Issue by robdaemon** [metabase/metabase#14560]
**Describe the bug**

Hosting specifies the DB connection URI in one env var, `MB_DB_CONNECTION_URI` - when running `dump-to-h2` this way, you get a schema error. This fails on `master`, and the `0.38.0-rc3` build.

**Logs**

```text
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
2021-01-27 17:39:58,889 INFO metabase.util :: Maximum memory available to JVM: 7.8 GB
2021-01-27 17:40:02,726 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓 
 For more information, see https://metabase.com/docs/latest/operations-guide/encrypting-database-details-at-rest.html
2021-01-27 17:40:05,916 INFO metabase.core :: 
Metabase null (null null) 

Copyright © 2021 Metabase, Inc. 

Metabase Enterprise Edition extensions are NOT PRESENT.
2021-01-27 17:40:05,922 WARN metabase.core :: WARNING: You have enabled namespace tracing, which could log sensitive information like db passwords.
Dumping from configured Metabase db to H2 file /tmp/a9819a38-0163-429e-a35f-c1cf86fe635f.db
clojure.lang.ExceptionInfo: Input to copy! does not match schema: 

	   [nil (named (not (matches-some-precondition? a-java.lang.String)) source-jdbc-spec) nil nil]  

 {:type :schema.core/error, :schema [#schema.core.One{:schema (enum :postgres :h2 :mysql), :optional? false, :name source-db-type} #schema.core.One{:schema (cond-pre #"^jdbc:" (pred "Valid map")), :optional? false, :name source-jdbc-spec} #schema.core.One{:schema (enum :postgres :h2 :mysql), :optional? false, :name target-db-type} #schema.core.One{:schema (cond-pre #"^jdbc:" (pred "Valid map")), :optional? false, :name target-jdbc-spec}], :value [:postgres "postgresql://hosting-8d3b7b373fbe43b3-rds.co0x6szdjg8b.us-west-2.rds.amazonaws.com:5432/metabase_hosting_8d3b7b373fbe43b3?user=metabase_hosting_8d3b7b373fbe43b3&password=8ojE6ifiSjUM58yyiwAntenZmfOtRKzo" :h2 {:classname "org.h2.Driver", :subprotocol "h2", :subname "file:/tmp/a9819a38-0163-429e-a35f-c1cf86fe635f.db"}], :error [nil (named (not (matches-some-precondition? a-java.lang.String)) source-jdbc-spec) nil nil]}
	at metabase.cmd.copy$fn__74568$copy_BANG___74573.invoke(copy.clj:256)
	at metabase.cmd.dump_to_h2$dump_to_h2_BANG_.invokeStatic(dump_to_h2.clj:36)
	at metabase.cmd.dump_to_h2$dump_to_h2_BANG_.invoke(dump_to_h2.clj:19)
	at clojure.lang.Var.invoke(Var.java:388)
	at metabase.cmd$dump_to_h2.invokeStatic(cmd.clj:54)
	at metabase.cmd$dump_to_h2.doInvoke(cmd.clj:46)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.core$apply.invoke(core.clj:660)
	at metabase.cmd$run_cmd$fn__74142.invoke(cmd.clj:171)
	at metabase.cmd$run_cmd.invokeStatic(cmd.clj:171)
	at metabase.cmd$run_cmd.invoke(cmd.clj:167)
	at clojure.lang.Var.invoke(Var.java:388)
	at metabase.core$run_cmd.invokeStatic(core.clj:148)
	at metabase.core$run_cmd.invoke(core.clj:146)
	at metabase.core$_main.invokeStatic(core.clj:170)
	at metabase.core$_main.doInvoke(core.clj:165)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at metabase.core.main(Unknown Source)
Command failed with exception: Input to copy! does not match schema: 

	   [nil (named (not (matches-some-precondition? a-java.lang.String)) source-jdbc-spec) nil nil]  
```

**To Reproduce**

Set up a postgres instance locally, and start Metabase on it.

```bash
export MB_DB_CONNECTION_URI='postgresql://localhost:5432/metabase?username=foo&password=foo'
java -jar metabase.jar
```

After the installation is complete, terminate this instance, and run the `dump-to-h2` command:

```bash
export MB_DB_CONNECTION_URI='postgresql://localhost:5432/metabase?username=foo&password=foo'
java -jar metabase.jar dump-to-h2 /tmp/mydump.db
```

**Comment by scott1979rn** [comment]
- [ ]

**Comment by camsaul** [comment]
Fixed by #14565
